dask/templates/additional-worker-deployment.yaml

```
@@ -0,0 +1,165 @@
+{{- range $worker := .Values.additional_worker_groups }}
```

When you use `range` it updates `.` to be the item you are iterating over, so you don't need to assign it to a variable. This also means we need to use `$` in place of `.` and `$.Foo` in place of `.Foo` throughout everything inside the range.

Also there is a `merge` function that deep merges two dictionaries. So we could take the additional worker dict and merge the default worker dict onto it instead of all the `if` `else` checking.

```suggestion
{{- range $.Values.additional_worker_groups }}
{{- $worker := merge . $.Values.worker }}
```

---

For example, would  `$worker.image.dask_worker` become `$.image.dask_worker`?

---

No `$worker` stays the same because it is a variable but `.Values` becomes `$.Values`.